test/test_client.py

```diff
@@ -461,3 +461,16 @@ async def test_additional_user_agent_components(client_session, url):
                 },
             }
         )
+
+
+async def test_pop_future_none(client_session, url, driver_ready):
+    """Test when a future has been cleared from futures dict, popping still works."""
+    client = Client(url, client_session)
+    await client.connect()
+
+    assert client.connected
+
+    asyncio.create_task(client.listen(driver_ready))
+
+    with pytest.raises(asyncio.CancelledError):
```

Comment:
Should we await `driver_ready` first?

Reply:
by awaiting `driver_ready` the connection is already closed by the time we try to send the command. Will have to review the fixture to see what the right implementation is here

Reply:
the problem is that as soon as we drain the message queue, we close the connection. We should instead only close the connection if we attempt to receive a message and the queue is already drained. Making some adjustments to the tests accordingly

Reply:
let me know what you think of these changes. I think this is the correct behavior, but I am not 100% sure